staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/server_test.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamiccertificates
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/client-go/util/cert"
+)
+
+func TestServingCert(t *testing.T) {
+	tlsConfig := &tls.Config{
+		// Can't use SSLv3 because of POODLE and BEAST
+		// Can't use TLSv1.0 because of POODLE and BEAST using CBC cipher
+		// Can't use TLSv1.1 because of RC4 cipher usage
+		MinVersion: tls.VersionTLS12,
+		// enable HTTP2 for go's 1.7 HTTP Server
+		NextProtos: []string{"h2", "http/1.1"},
+	}
+
+	defaultCertProvider, err := createTestTLSCerts(testCertSpec{
+		host:  "172.30.0.1",
+		ips:   []string{"172.30.0.1"},
+		names: []string{"kubernetes", "kubernetes.default.svc.cluster.local"},
+	}, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var sniCerts []SNICertKeyContentProvider
+	certSpecs := []testCertSpec{
+		{
+			host:  "172.30.0.1",
+			ips:   []string{"172.30.0.1"},
```

Comment:
can we add an IPv6 test case?
this test worked for me locally
```
diff --git a/staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/server_test.go b/staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/server_test.go
index 8f0780440e..c8da0eb9df 100644
--- a/staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/server_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/server_test.go
@@ -61,6 +61,11 @@ func TestServingCert(t *testing.T) {
                        ips:   []string{"127.0.0.1"},
                        names: []string{"localhost"},
                },
+               {
+                       host:  "2001:abcd:bcda::1",
+                       ips:   []string{"2001:abcd:bcda::1"},
+                       names: []string{"openshiftv6", "openshiftv6.default.svc.cluster.local"},
+               },
        }
 
        for _, certSpec := range certSpecs {
@@ -114,6 +119,7 @@ func TestServingCert(t *testing.T) {
        expectedDefaultCertBytes, _ := defaultCertProvider.CurrentCertKeyContent()
        expectedServiceCertBytes, _ := sniCerts[0].CurrentCertKeyContent()
        expectedLocalhostCertBytes, _ := sniCerts[1].CurrentCertKeyContent()
+       expectedServiceV6CertBytes, _ := sniCerts[2].CurrentCertKeyContent()
 
        tests := []struct {
                name       string
@@ -135,6 +141,11 @@ func TestServingCert(t *testing.T) {
                        serverName: "openshift",
                        expected:   []byte(strings.TrimSpace(string(expectedServiceCertBytes))),
                },
+               {
+                       name:       "service v6 by dns",
+                       serverName: "openshiftv6",
+                       expected:   []byte(strings.TrimSpace(string(expectedServiceV6CertBytes))),
+               },
                {
                        name:       "localhost by dns",
                        serverName: "localhost",
```

Reply:
+1 for ipv6 test

Reply:
added